spec/unit/roles_create_spec.rb

```diff
@@ -67,6 +66,30 @@ def self.now
       deploy = @connection.deploys.create(params)
       expect(readFile('create_role.xml')).to eq(@receivedXML)
     end
+
+    describe 'when tcp endpoints are present in mappings' do
+      it 'assigns the tcp endpoint name from the mappings' do
```

Comment:
Can we add a test here that explicitly validates that the port names are in a particular format? That way the name format is documented and we can also avoid regressions.